pgjdbc/src/test/java/org/postgresql/test/jdbc2/TimestamptzTest.java

```diff
@@ -0,0 +1,239 @@
+    }
+    return ids;
+  }
+
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    TestUtil.createSchema(con, "testtimestamp");
+    TestUtil.createTable(con, "testtimestamp.tbtesttimestamp", "id bigint, ts timestamptz");
+  }
+
+  @Override
+  public void tearDown() throws SQLException {
+    TestUtil.dropTable(con, "testtimestamp.tbtesttimestamp");
+    TestUtil.dropSchema(con, "testtimestamp");
+    super.tearDown();
+  }
+
+  /**
+   * Aother example where overloading is necessary on sql site to choose the correct function
+   */
+  @Test
+  public void testTypeOnDbSite_functionOverloading() throws SQLException {
+
+    //SET time zone 'Europe/Berlin';
+    try (PreparedStatement ps = con.prepareStatement("create or replace function testtimestamp.demoF(msg text, ts timestamptz) returns text language sql as $$select msg || ' at time ' || ts $$;  ")) {
+      ps.executeUpdate();
+    }
+    try (PreparedStatement ps = con.prepareStatement("create or replace function testtimestamp.demoF(msg text, additional text) returns text language sql as $$select msg || ' wrongInfo:' || additional  $$; ")) {
+      ps.executeUpdate();
+    }
+
+    GregorianCalendar cal = new GregorianCalendar(TimeZone.getTimeZone("Europe/Berlin"));
+    cal.set(2023, Calendar.MARCH, 12, 9, 30);
+    cal.set(Calendar.SECOND, 0);
+    cal.set(Calendar.MILLISECOND, 0);
+
+    try (PreparedStatement ps = con.prepareStatement(" SELECT  testtimestamp.demoF(?, ?) as res ")) {
+      ps.setString(1, "Some data");
+      ps.setTimestamp(2, new Timestamp(cal.getTimeInMillis()));
+
+      ResultSet rs = ps.executeQuery();
+      if (rs.next()) {
+        String resultFunction = rs.getString("res");
+        System.out.printf("result:%s \n", resultFunction);
+
+        if (resultFunction.contains("wrongInfo")) {
+          fail("choose wrong overloading");
+        }
+
+      } else {
+        fail("no result");
+      }
+    }
+  }
+
+
+
+  /**
+   * Test to demonstrate another example on additional unnecessary cast.
+   * The driver know already the data type.
+   */
+  @Test
+  public void testTypeOnDbSite_select() throws SQLException {
+
+    //SET time zone 'Europe/Berlin';
+    try (PreparedStatement ps = con.prepareStatement(" insert into testtimestamp.tbtesttimestamp (id, ts) values(1, '2023-03-12 10:00:00+1'::timestamptz ) ")) {
+      ps.executeUpdate();
+    }
+
+    GregorianCalendar cal = new GregorianCalendar(TimeZone.getTimeZone("Europe/Berlin"));
+    cal.set(2023, Calendar.MARCH, 12, 9, 30);
+    cal.set(Calendar.SECOND, 0);
+    cal.set(Calendar.MILLISECOND, 0);
+
+    try (PreparedStatement ps = con.prepareStatement(" SELECT * from testtimestamp.tbtesttimestamp where ts < ?  +  interval '1 hour' ")) {
+      ps.setTimestamp(1, new Timestamp(cal.getTimeInMillis()));
+
+      ResultSet rs = ps.executeQuery();
+      if (rs.next()) {
+        System.out.printf("id:%d ts:%s \n", rs.getInt("id"), rs.getTimestamp("ts"));
```

Comment:
This should assert the correct values here

Reply:
done